Modules/Core/Common/include/itkMacro.h

```diff
@@ -1341,16 +1343,231 @@ compilers.
+{
+public:
+  // Inherit the constructors from its base class.
+  using ExceptionObject::ExceptionObject;
+
+  /** \see LightObject::GetNameOfClass() */
+  itkOverrideGetNameOfClassMacro(MemoryAllocationError);
+};
+
+/** \class RangeError
+ * Exception thrown when accessing memory out of range.
+ * \ingroup ITKSystemObjects
+ * \ingroup ITKCommon
+ */
+class ITKCommon_EXPORT RangeError : public ExceptionObject
+{
+public:
+  // Inherit the constructors from its base class.
+  using ExceptionObject::ExceptionObject;
+
+  /** \see LightObject::GetNameOfClass() */
+  itkOverrideGetNameOfClassMacro(RangeError);
+};
+
+/** \class InvalidArgumentError
+ * Exception thrown when invalid argument is given to a method
+ * or function.
+ * \ingroup ITKSystemObjects
+ * \ingroup ITKCommon
+ */
+class ITKCommon_EXPORT InvalidArgumentError : public ExceptionObject
+{
+public:
+  // Inherit the constructors from its base class.
+  using ExceptionObject::ExceptionObject;
+
+  /** \see LightObject::GetNameOfClass() */
+  itkOverrideGetNameOfClassMacro(InvalidArgumentError);
+};
+
+/** \class IncompatibleOperandsError
+ * Exception thrown when two operands are incompatible.
+ * \ingroup ITKSystemObjects
+ * \ingroup ITKCommon
+ */
+class ITKCommon_EXPORT IncompatibleOperandsError : public ExceptionObject
+{
+public:
+  // Inherit the constructors from its base class.
+  using ExceptionObject::ExceptionObject;
+
+  /** \see LightObject::GetNameOfClass() */
+  itkOverrideGetNameOfClassMacro(IncompatibleOperandsError);
+};
+
+/** \class ProcessAborted
+ * Exception thrown when a filter (actually a ProcessObject) has been aborted.
+ * \ingroup ITKSystemObjects
+ * \ingroup ITKCommon
+ */
+class ITKCommon_EXPORT ProcessAborted : public ExceptionObject
+{
+public:
+  /** Default constructor.  Needed to ensure the exception object can be
+   * copied. */
+  ProcessAborted()
+    : ExceptionObject()
+  {
+    this->SetDescription("Filter execution was aborted by an external request");
+  }
+
+  /** Constructor. Needed to ensure the exception object can be copied. */
+  ProcessAborted(const char * file, unsigned int lineNumber)
+    : ExceptionObject(file, lineNumber)
+  {
+    this->SetDescription("Filter execution was aborted by an external request");
+  }
+
+  /** Constructor. Needed to ensure the exception object can be copied. */
+  ProcessAborted(const std::string & file, unsigned int lineNumber)
+    : ExceptionObject(file, lineNumber)
+  {
+    this->SetDescription("Filter execution was aborted by an external request");
+  }
+
+  /** \see LightObject::GetNameOfClass() */
+  itkOverrideGetNameOfClassMacro(ProcessAborted);
+};
+} // end namespace itk
+// ====== End itkExceptionObject declaration
 
 /** itkDynamicCastInDebugMode
  * Use static_cast in Release builds, and dynamic_cast in Debug
  *
  * Note: this must come after:
- *
- *   #include "itkExceptionObject.h"
+ *  defining itkExceptionObject.h above
```

Comment:
Can we add a historical note here that itkExceptionObject.h was previously separate file but itkMacro.h should only be #include's regardless?

Reply:
yes.  I'll add that historical note at the start of the location where the old include used to be.